src/scripts/main.js

```diff
@@ -1,3 +1,211 @@
+  <label> 
+    Name:
+      <input name="name" type="text" data-qa="name" required>
+  </label>
+  <label> 
+    Position:
+      <input name="position" type="text" data-qa="position" required>
+  </label>
+  <label> 
+    Office:
+      <select name="office" type="text" data-qa="office" required>
+      <option value="Tokyo">Tokyo</option>
+      <option value="Singapore">Singapore</option>
+      <option value="London">London</option>
+      <option value="New York">New York</option>
+      <option value="Edinburgh">Edinburgh</option>
+      <option value="San francisco">San Francisco</option>
+      </select>
+  </label>
+  <label> 
+    Age:
+      <input name="age" type="number" required>
+  </label>
+  <label> 
+    Salary:
+      <input name="salary" type="number" required>
+  </label>
+  <button type="submit">Save to table</button>
+  `;
+
+newEmployeeForm.addEventListener('submit', (evt) => {
+  evt.preventDefault();
+
+  const data = Object.fromEntries(new FormData(newEmployeeForm));
+  const employee = document.createElement('tr');
+
+  employee.innerHTML = `
+    <tr>
+      <td>${data.name}</td>
+      <td>${data.position}</td>
+      <td>${data.office}</td>
+      <td>${data.age}</td>
+      <td>${'$' + (+data.salary).toLocaleString('en-US')}</td>
+    </tr>
+  `;
+
+  if (data.name.length < 4) {
+    pushNotification('Error', 'Name should have more than 3 letters');
+  } else if (data.age < 18 || data > 90) {
+    pushNotification('Error', 'Incorrect age', 'error');
+  } else {
+    body.append(employee);
+    pushNotification('Success', 'Employee added successfully', 'success');
+  }
+
+  newEmployeeForm.reset();
+});
+
+body.addEventListener('dblclick', (evt) => {
+  const selectedCell = evt.target;
+  const previousValue = selectedCell.textContent;
+  const newCell = document.createElement('input');
+
+  newCell.classList.add('cell-input');
+
+  switch (selectedCell.cellIndex) {
+    case 3:
+      newCell.setAttribute('type', 'number');
+
+      break;
+
+    default:
+      newCell.setAttribute('type', 'text');
+
+      break;
+  }
+
+  selectedCell.textContent = '';
+
+  selectedCell.append(newCell);
+  newCell.focus();
+
+  const modifyCell = function() {
+    switch (selectedCell.cellIndex) {
+      case 0:
+        if (newCell.value.length < 4) {
+          pushNotification('Error', 'Name should have more than 3 letters');
+          newCell.value = previousValue;
+        }
+
+        break;
+
+      case 3:
+        if (newCell.value < 18 || newCell.value > 90) {
+          pushNotification('Error', 'Incorrect age', 'error');
+          newCell.value = previousValue;
+        }
+
+        break;
+
+      case 4:
+        newCell.value = newCell.value !== ''
+          ? '$' + Number(newCell.value.remove('$')).toLocaleString('en-US')
+          : previousValue;
+
+        break;
+
+      default:
+        if (newCell.value === '') {
+          newCell.value = previousValue;
+        }
+
+        break;
+    }
```

Comment:
It's a very complex block for me. 
Could you use only one approach (`switch/case` or `if/else`)?